merlin/models/tf/core.py

```diff
@@ -2491,6 +2627,12 @@ def fit(
 
             x = Dataset(x, batch_size=batch_size, **kwargs)
 
+        if callbacks is None:
+            callbacks = []
+
+        # Adding a callback to control metrics computation
+        callbacks.append(MetricsComputeCallback(train_metrics_steps))
```

Comment:
We probably should check whether `MetricsComputeCallback` is already part of the callbacks.